tests/integration/test_auth_client_flow.py

```diff
@@ -0,0 +1,133 @@
+                         "response_type=" + "code",
+                         "code_challenge=" + quote_plus(remade_challenge),
+                         "code_challenge_method=" + "S256",
+                         "access_type=" + "offline"]
+        for val in expected_vals:
+            self.assertIn(val, url_res)
+
+    def test_oauth2_get_authorize_url_confidential(self):
+        """
+        Starts an auth flow with a NativeAppFlowManager, gets the authorize url
+        validates expected results with both default and specified parameters.
+        """
+        ac = globus_sdk.AuthClient(
+            client_id=get_client_data()["confidential_app_client1"]["id"])
+
+        # default parameters for starting auth flow
+        flow_manager = globus_sdk.auth.GlobusAuthorizationCodeFlowManager(
+            ac, "uri")
+        ac.current_oauth2_flow_manager = flow_manager
+
+        # get url_and validate results
+        url_res = ac.oauth2_get_authorize_url()
+        expected_vals = [ac.base_url + "v2/oauth2/authorize?",
+                         "client_id=" + ac.client_id,
+                         "redirect_uri=" + "uri",
+                         "scope=" + quote_plus(DEFAULT_REQUESTED_SCOPES),
+                         "state=" + "_default",
+                         "response_type=" + "code",
+                         "access_type=" + "online"]
+        for val in expected_vals:
+            self.assertIn(val, url_res)
+
+        # starting flow with specified paramaters
+        flow_manager = globus_sdk.auth.GlobusAuthorizationCodeFlowManager(
+            ac, requested_scopes="scopes", redirect_uri="uri",
+            state="state", refresh_tokens=True)
+        ac.current_oauth2_flow_manager = flow_manager
+
+        # get url_and validate results
+        url_res = ac.oauth2_get_authorize_url()
+        expected_vals = [ac.base_url + "v2/oauth2/authorize?",
+                         "client_id=" + ac.client_id,
+                         "redirect_uri=" + "uri",
+                         "scope=" + "scopes",
+                         "state=" + "state",
+                         "response_type=" + "code",
+                         "access_type=" + "offline"]
+        for val in expected_vals:
+            self.assertIn(val, url_res)
+
+    def test_oauth2_exchange_code_for_tokens_native(self):
+        """
+        Starts a NativeAppFlowManager, Confirms invalid code raises 401
+        Further testing cannot be done without user login credentials
+        """
+        ac = globus_sdk.AuthClient(
+            client_id=get_client_data()["native_app_client1"]["id"])
+        flow_manager = globus_sdk.auth.GlobusNativeAppFlowManager(ac)
+        ac.current_oauth2_flow_manager = flow_manager
+
+        with self.assertRaises(GlobusAPIError) as apiErr:
+            ac.oauth2_exchange_code_for_tokens("invalid_code")
+        self.assertEqual(apiErr.exception.http_status, 401)
+        self.assertEqual(apiErr.exception.code, "Error")
```

Comment:
I know you expressed some concern over the fact that we can only test for errors on these, not a real code exchange.
This is fine for now.

If we want to test a successful code exchange, we need to bite off the work of automating a login flow in the browser. We decided to skip on that for now, so this is the best we can do for today.